drivers/azure/azureutil/azureutil.go

```diff
@@ -520,6 +534,32 @@ func (a AzureClient) CreateVirtualMachine(resourceGroup, name, location, size, a
 	return err
 }
 
+func (a AzureClient) CreateVirtualMachineExtension(os, resourceGroup, name, location string) error {
+	vmExtension := compute.VirtualMachineExtension{
```

Comment:
can we add here a logging statement like the following:

    Adding CustomScriptExtension to virtual machine

otherwise the users won't be able to see the progress or feel that it is proceeding very clearly.